File: trip-kit-booking/src/main/java/com/skedgo/android/tripkit/booking/ExternalOAuth.java
package com.skedgo.android.tripkit.booking;

import androidx.annotation.Nullable;

import org.immutables.value.Value;

import java.util.UUID;

@Value.Immutable
public abstract class ExternalOAuth {

  public abstract String authServiceId();
  public abstract String token();
  @Nullable public abstract String refreshToken();
  public abstract long expiresIn();

  @Value.Default public String id() {
    return UUID.randomUUID().toString();
  }

}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/ExternalOAuth.kt
package com.skedgo.tripkit.booking

import org.immutables.value.Value.Default
import org.immutables.value.Value.Immutable
import java.util.UUID

@Immutable
abstract class ExternalOAuth {
    abstract fun authServiceId(): String

    abstract fun token(): String

    abstract fun refreshToken(): String?

    abstract fun expiresIn(): Long

    @Default
    open fun id(): String {
        return UUID.randomUUID().toString()
    }
}
